libsolidity/analysis/ControlFlowBuilder.cpp

```diff
@@ -85,6 +85,21 @@ bool ControlFlowBuilder::visit(Conditional const& _conditional)
 	return false;
 }
 
+bool ControlFlowBuilder::visit(TryStatement const& _tryStatement)
+{
+	appendControlFlow(_tryStatement.externalCall());
+
+	auto nodes = splitFlow(_tryStatement.clauses().size());
+	for (size_t i = 0; i < _tryStatement.clauses().size(); ++i)
+	{
+		m_currentNode = nodes[i];
+		appendControlFlow(_tryStatement.clauses()[i]->block());
+	}
+	mergeFlow(nodes);
```

Comment:
Shouldn't this rather merge the blocks that are appended?

Reply:
You are right! Do you think a `nodes[i] = m_currentNode;` at the end of the loop body would work?

Reply:
I think so

Reply:
Hm, it should actually fail this test: https://github.com/ethereum/solidity/pull/7328/files#diff-78f7d7bd8318b35cb1e6168f79a3153d